modules/cockroachdb/cockroachdb.go

```diff
@@ -0,0 +1,259 @@
+package cockroachdb
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"net"
+	"net/url"
+	"path/filepath"
+
+	"github.com/docker/go-connections/nat"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/stdlib"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+var ErrTLSNotEnabled = fmt.Errorf("tls not enabled")
+
+const (
+	certsDir = "/tmp"
+
+	defaultSQLPort   = "26257/tcp"
+	defaultAdminPort = "8080/tcp"
+
+	defaultImage     = "cockroachdb/cockroach:latest-v23.1"
+	defaultUser      = "root"
+	defaultPassword  = ""
+	defaultDatabase  = "defaultdb"
+	defaultStoreSize = "100%"
+)
+
+// CockroachDBContainer represents the CockroachDB container type used in the module
+type CockroachDBContainer struct {
+	testcontainers.Container
+	opts options
+}
+
+// MustConnectionString panics if the address cannot be determined.
+func (c *CockroachDBContainer) MustConnectionString(ctx context.Context) string {
+	addr, err := c.ConnectionString(ctx)
+	if err != nil {
+		panic(err)
+	}
+	return addr
+}
+
+// ConnectionString returns the dial address to open a new connection to CockroachDB.
+func (c *CockroachDBContainer) ConnectionString(ctx context.Context) (string, error) {
+	port, err := c.MappedPort(ctx, defaultSQLPort)
+	if err != nil {
+		return "", err
+	}
+
+	host, err := c.Host(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	return connString(c.opts, host, port), nil
+}
+
+// TLSConfig returns config necessary to connect to CockroachDB over TLS.
+func (c *CockroachDBContainer) ConnectionTLS() (*tls.Config, error) {
```

Comment:
I think this name would be even clearer. If you agree on that (the comment indicates you already thought about that 😅), let's update the docs too.
```suggestion
func (c *CockroachDBContainer) TLSConfig() (*tls.Config, error) {
```